operator/standalone/delete_it_test.go

```diff
@@ -166,16 +145,6 @@ func (ts *DeleteStandalonePipelineSuite) Test_DeleteConnectionSecret() {
 			givenSecret:    "existing-secret",
 			expectedError:  "",
 		},
-		"GivenExistingSecret_WhenDeletingGeneratesAnError_ThenReturnError": {
-			prepare: func(name, namespace string) {
-				ts.EnsureNS(namespace)
-				ts.EnsureResources(&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "secret", Namespace: namespace}})
-			},
-			givenNamespace: "test-namespace",
-			// we purposefully set namespace to empty so that we generate an error so that we avoid mocking.
-			givenSecret:   "",
-			expectedError: "resource name may not be empty",
-		},
```

Comment:
@A note on why I removed this and similar test cases:
Previously, it was tested whether the function under test returns an error if the Delete() func from the client returned an error. We did trigger this error by not setting a name, which returns "resource name may not be empty". But not setting a name is basically an exploit of the exact bug that I'm about to fix in this PR. Thus, I don't see any other easy way to trigger an error without getting involved in mocking or using the fake client.

However, I don't see covering the Delete() func from the client worth to test at this stage right now. The more important logic is already covered and can be verified.

Reply:
For me it's fine to remove it. If we want to do proper testing we need mocks and end-to-end tests. This specific test case is just a coincidence to what happened in the real world scenario which by the way should have been caught with an end-to-end test.